scripts/o3de/o3de/repo_properties.py

```diff
@@ -306,12 +306,15 @@ def edit_repo_props(repo_path: pathlib.Path = None,
         repo_json['repo_name'] = repo_name
 
     if add_gems or delete_gems or replace_gems:
+        utils.backup_file(repo_path)
```

Comment:
In the testing framework for the O3DE CLI, we have unit tests outlined for utils.py, as seen in this path: scripts/o3de/tests/test_utils.py

I think the `backup_file` method could warrant a unit test to verify its intended behavior. Currently that is missing from test_utils.py. That may be out of scope for the current PR, but something to consider.

Reply:
yeah, out of scope for this PR, it would be more relevant for this PR to create unit tests for `repo_properties` and mock the `utils.backup_file` instead. 
In other news, if we decide to eventually unit test 'backup_file', make sure to use pytest's temp directory as it is automatically cleaned up for you

https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html

Reply:
Thank you T.J and Steve!! Will definitely look into this